x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/mappings_editor.tsx

```diff
@@ -24,37 +25,78 @@ interface Props {
   indexSettings?: IndexSettings;
 }
 
+type TabName = 'fields' | 'advanced';
+
 export const MappingsEditor = React.memo(
   ({ onUpdate, defaultValue = {}, indexSettings }: Props) => {
-    const [selectedTab, selectTab] = useState<'fields' | 'advanced'>('fields');
-
-    const {
-      _source = {},
-      dynamic,
-      numeric_detection,
-      date_detection,
-      dynamic_date_formats,
-      properties = {},
-    } = defaultValue;
-
-    const { configurationDefaultValue, fieldsDefaultValue } = useMemo(
-      () => ({
-        configurationDefaultValue: {
+    const [selectedTab, selectTab] = useState<TabName>('fields');
+
+    const parsedDefaultValue = useMemo(() => {
+      const {
+        _source = {},
+        dynamic,
+        numeric_detection,
+        date_detection,
+        dynamic_date_formats,
+        properties = {},
+      } = defaultValue;
+
+      return {
+        configuration: {
           _source,
           dynamic,
           numeric_detection,
           date_detection,
           dynamic_date_formats,
         },
-        fieldsDefaultValue: properties,
-      }),
-      [defaultValue]
-    );
+        fields: properties,
+      };
+    }, [defaultValue]);
+
+    const validateConfigurationForm = async (
+      configuration: State['configuration']
+    ): Promise<{ isValid: boolean; data?: MappingsConfiguration }> => {
+      const isValid = await configuration.validate();
+
+      if (!isValid) {
+        return { isValid };
+      }
+
+      // We create a snapshot of the configuration form data.
+      const formData = configuration.data.format();
+
+      return { isValid, data: formData };
+    };
+
+    const changeTab = async (tab: TabName, [state, dispatch]: [State, Dispatch]) => {
+      if (tab === 'fields') {
```

Comment:
Would it make more sense here to check the value of `selectedTab`, to make it clear that we're concerned about persisting the state of the _advanced_ tab?

```js
if (selectedTab === 'advanced') {
  // ...
}
```
```

Reply:
Good point! I will make that change on my upcoming PR.